include/swift/Basic/LangOptions.h

```diff
@@ -171,6 +171,9 @@ namespace swift {
     /// Emit a remark after loading a module.
     bool EnableModuleLoadingRemarks = false;
 
+    /// Whether to serialize comments for SPI symbols.
+    bool EmitCommentsForSPISymbols = false;
+
```

Comment:
Generally, LangOptions is for things that affect the language mode/features and things like warnings/errors/remarks which get emitted, so it seems a bit odd to have a serialization related option here. Maybe in the future, there should be a separate struct to keep track of what options should be used for symbol graphs/swift doc files? We do have IRGenOptions, so this would be similar, since it is another output of the compiler.

Reply:
There's a `SymbolGraphOptions` struct already, but IIRC it's currently only generated and used specifically in the symbol-graph code. Would it be worth using this here?

Reply:
Oh, I didn't realize that. I see there is a

```
/// Whether to emit symbols with SPI information.
bool IncludeSPISymbols;
```

field on that struct, so that does seem like the right place for this flag, if the plumbing is not too much work. Otherwise, it's also OK by me if you leave a TODO here or something to indicate that this should be changed/fixed.